lib/github/merge/local.ex

```diff
@@ -0,0 +1,229 @@
+defmodule BorsNG.GitHub.Merge.Local do
+  @moduledoc """
+  Uses local Git to construct staging/trying branches instead of GitHub's
+  API.
+
+  Only used when local merges are enabled.
+  """
+
+  use GenServer
+
+  alias BorsNG.Database.Repo
+  alias BorsNG.Database.Batch
+  alias BorsNG.Database.Project
+  alias BorsNG.Database.LinkPatchBatch
+  alias BorsNG.GitHub
+  alias BorsNG.GitHub.Merge.Hooks
+  alias BorsNG.Worker.Batcher
+
+  def start_link do
+    GenServer.start_link(__MODULE__, :ok, name: __MODULE__)
+  end
```

Comment:
about GenServer: this is an Erlang-y/Elixir-y idiom. rather than this being a module that you call library functions from, this module is actually a separate (lightweight Erlang) process that receives messages from other processes. if it crashes or exceptions happen, this process will automatically get restarted without affecting the rest of the application. it doesn't change the code that much, but the main thing this enables is that exceptions or pattern match failures aren't as important, since it won't cause loss of service.

importantly, the code is written in such a way that there's only one process based off this module running at any given time, since we don't want multiple things modifying the filesystem at the same time.